### PR TITLE
refactor: monorepo (use yarn workspace)

### PR DESCRIPTION
**Pull request by aarongranick-okta**
Main module source is moved to `packages/okta-auth-js`

3 support modules are also created under `test`:
  - `@okta/test.app` Used by karma
  - `@okta/test.support` formerly test/util (also includes test/xhr). Support code used by tests.
  - `@okta/test.e2e`

### CODE REVIEW

**Comment by aarongranick-okta** [comment]
@haishengwu-okta @swiftone I have moved the test app, support and e2e suite to their own namespace under packages, `@test`. I also did some fixes/tightening on the lint & some minor hardening of the e2e tests.

File: package.json [hunk 1]
```diff
@@ -1,83 +1,32 @@
...
+    "prepare": "yarn workspace @okta/okta-auth-js build",
+    "start": "yarn workspace @test/app start --open"
   },
-  "jest-junit": {
-    "output": "./build2/reports/unit/junit-result.xml"
-  }
+  "workspaces": [
+    "packages/*",
+    "packages/@okta/*",
+    "packages/@test/*"
```
Comment (swiftone) [inline package.json]: I'm not a fan of creating our own fake namespace - is this a conventional practice?

File: packages/@okta/okta-auth-js/package.json [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  "okta-auth-js": {
+    "STATE_TOKEN_KEY_NAME": "oktaStateToken",
+    "DEFAULT_POLLING_DELAY": 500,
+    "DEFAULT_MAX_CLOCK_SKEW": 300,
+    "DEFAULT_CACHE_DURATION": 86400,
+    "FRAME_ID": "okta-oauth-helper-frame",
+    "REDIRECT_OAUTH_PARAMS_COOKIE_NAME": "okta-oauth-redirect-params",
+    "REDIRECT_STATE_COOKIE_NAME": "okta-oauth-state",
+    "REDIRECT_NONCE_COOKIE_NAME": "okta-oauth-nonce",
+    "TOKEN_STORAGE_NAME": "okta-token-storage",
```
Comment (swiftone) [inline packages/@okta/okta-auth-js/package.json]: IIRC, this isn't used, so orthogonal to your discussion on whether to move these, I'd recommend ripping out the ones that don't exist anymore.

File: packages/@okta/okta-auth-js/test/spec/factor-enroll.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
 jest.mock('cross-fetch');
 
-var util = require('../util/util'),
+var util = require('@test/support/util'),
```
Comment (swiftone) [inline packages/@okta/okta-auth-js/test/spec/factor-enroll.js]: nit: I much prefer having each variable individually declared so we don't have to skim more than the relevant line, which is also consistent with the rest of the repo.

File: packages/@okta/okta-auth-js/package.json (hunk 2 above)
Comment (aarongranick-okta) [reply packages/@okta/okta-auth-js/package.json]: 👍 This refactor includes the generated `lib/config.js` checked in. I have added an issue for this work: https://oktainc.atlassian.net/browse/OKTA-259362

File: package.json (hunk 1 above)
Comment (aarongranick-okta) [reply package.json]: As we discussed, I will remove the `@` prefix for test

**Comment by aarongranick-okta** [comment]
@swiftone I have moved all test modules to a folder named `test` located at the root of the project.

File: packages/okta-auth-js/test/spec/mfa-challenge.js [hunk 4]
```diff
@@ -1,6 +1,14 @@
 jest.mock('cross-fetch');
 
-var util = require('../util/util');
+var util = require('@okta/test.support/util');
```
Comment (haishengwu-okta) [inline packages/okta-auth-js/test/spec/mfa-challenge.js]: `util` is imported but not used. is that intentional?

File: packages/okta-auth-js/test/spec/mfa-enroll-activate.js [hunk 5]
```diff
@@ -1,6 +1,14 @@
 jest.mock('cross-fetch');
 
-var util = require('../util/util');
+var util = require('@okta/test.support/util');
+var Q = require('q');
+
+function mockQDelay() {
```
Comment (haishengwu-okta) [inline packages/okta-auth-js/test/spec/mfa-enroll-activate.js]: ditto. why write another mock method instead of using `util.mockQDelay`

Comment (aarongranick-okta) [reply packages/okta-auth-js/test/spec/mfa-enroll-activate.js]: In this case we are mocking a library that the test/lib depends on (Q), I do not want to add Q as a dependency of test-support, we are phasing it out.

File: packages/okta-auth-js/test/spec/mfa-challenge.js (hunk 4 above)
Comment (aarongranick-okta) [reply packages/okta-auth-js/test/spec/mfa-challenge.js]: It is used in the file, but there is no diff.

**Pull request merged**